src/Workspaces/Core/Portable/CodeStyle/CodeStyleHelpers.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    internal static class CodeStyleHelpers
+    {
+        public static CodeStyleOption<bool> ParseEditorConfigCodeStyleOption(string arg)
+        {
+            var args = arg.Split(':');
+            bool isEnabled = false;
+            if (args.Length != 2)
+            {
+                if (arg.Length == 1)
+                {
+                    if (bool.TryParse(args[0].Trim(), out isEnabled) && !isEnabled)
+                    {
+                        return new CodeStyleOption<bool>(value: isEnabled, notification: NotificationOption.None);
+                    }
+                    else
+                    {
+                        return CodeStyleOption<bool>.Default;
+                    }
+                }
+                return CodeStyleOption<bool>.Default;
```

Comment:
The semantics for these editorconfig files is that if the user has specified the option, but we can't parse it, we pass the default option.  Would you prefer we throw?

---
In reply to: [87657312](https://github.com/dotnet/roslyn/pull/15029#discussion_r87657312) [](ancestors = 87657312)